src/effects/VST/VSTEffect.cpp

```diff
@@ -1193,6 +1187,48 @@ bool VSTEffect::SaveSettings(const EffectSettings& settings, CommandParameters&
 }
 
 
+bool VSTEffect::CopySettingsContents(const EffectSettings& src, EffectSettings& dst, SettingsCopyDirection copyDirection) const
+{
+   const VSTEffectSettings* pSrc = src.cast<VSTEffectSettings>();
+         VSTEffectSettings* pDst = dst.cast<VSTEffectSettings>();
+
+   pDst->mNumParams = pSrc->mNumParams;
+   pDst->mUniqueID  = pSrc->mUniqueID;
+   pDst->mVersion   = pSrc->mVersion;
+
+   assert(pDst->mParamsMap.size() == pSrc->mParamsMap.size());
+
+   // TODO
+   // Without this assignment, changes in sound will not be heard when
+   // realtime processing and moving a slider (for plugins which use the chunk).
+   // 
+   // We must be sure that this assignment does not create a memory allocation -
+   // it is not allowed in the thread which executes this.
+   //
+   // If this could allocate, we must disable it and somehow force effects
+   // which use the chunk, to use the fallback key-value map instead.
+   //
+   pDst->mChunk = pSrc->mChunk;
+
+   // Do an in-place rewrite of dst, avoiding allocations
+   auto& dstMap = pDst->mParamsMap;
+   auto dstIter = dstMap.begin(), dstEnd = dstMap.end();
+   const auto& srcMap = pSrc->mParamsMap;
+
+   for (auto& [key, value] : srcMap)
+   {
+      assert(dstIter != dstEnd);
+      auto& [dstKey, dstValue] = *dstIter;
+      assert(dstKey == key);
+
+      dstValue = value;
+
+      dstIter++;
+   }
```

Comment:
One more assertion after the loop, that dstIter is the end iter, would complete the assertion of correspondence of keys

Reply:
ok, done